Set/Models/GameModel.swift

```diff
@@ -122,20 +148,23 @@ struct GameModel {
         }
     }
     
-    mutating private func getBonus(for player: Player) {
-        if player.id == firstPlayer.id {
-            firstPlayer.increaseScore(by: Constants.defaultBonus)
-        } else if player.id == secondPlayer.id{
-            secondPlayer.increaseScore(by: Constants.defaultBonus)
-        }
+    private func currentTurnDuration(for turnDate: Date) -> TimeInterval {
+        return turnDate.timeIntervalSince(previousSuccesfullMatchDate ?? startGameDate)
     }
     
-    mutating private func getPenalty(for player: Player) {
-        if player.id == firstPlayer.id, firstPlayer.score != 0 {
-            firstPlayer.decreaseScore(by: Constants.defaultPenalty)
-        } else if player.id == secondPlayer.id, secondPlayer.score != 0 {
-            secondPlayer.decreaseScore(by: Constants.defaultPenalty)
+    private func calculateBonus(for timeDuration: TimeInterval) -> Int {
+        var bonus: Int = 0
+        
+        if let previousTurnDuration = previousTurnDuration {
+            if timeDuration < previousTurnDuration {
+                bonus = Constants.defaultBonus + 1
```

Comment:
You can move extra bonus to constants too

Reply:
done